mlflow/store/sqlalchemy_store.py

```diff
@@ -564,3 +566,52 @@ def log_batch(self, run_id, metrics, params, tags):
             raise e
         except Exception as e:
             raise MlflowException(e, INTERNAL_ERROR)
+
+    def sample_oldest_metrics(self, timestamp, ratio, last_execution):
+        with self.ManagedSessionMaker() as session:
+            if last_execution:
+                runs = session.query(SqlRun.run_uuid)\
+                    .filter(SqlRun.start_time < timestamp, SqlRun.start_time > last_execution).all()
+            else:
+                runs = session.query(SqlRun.run_uuid).filter(SqlRun.start_time < timestamp).all()
+            runs = list(map(lambda x: x[0], runs))
+            metrics_query = session.query(SqlMetric).filter(SqlMetric.run_uuid.in_(runs))
+            metrics = pd.read_sql(metrics_query.statement, metrics_query.session.bind)
+            sampled_metrics = self._sample_metrics(metrics, runs, ratio)
+            session.query(SqlMetric).filter(SqlMetric.run_uuid.in_(runs))\
+                .delete(synchronize_session=False)
+            session.commit()
+        sampled_metrics.to_sql('metrics', self.engine, if_exists='append', index=False)
+
+    def _sample_metrics(self, metrics, runs, ratio):
+        result_df = pd.DataFrame()
+        for run in runs:
+            df_single_run = metrics.loc[metrics["run_uuid"] == run]
+            unique_metrics = df_single_run["key"].unique().tolist()
+            for metric in unique_metrics:
+                steps = df_single_run[df_single_run["key"] == metric]["step"].unique().tolist()
+                number_of_steps = int(len(steps) * ratio + 1)
+                selected_steps = np.linspace(0, len(steps) - 1, num=number_of_steps, dtype=int)
+                steps_to_keep = [steps[i] for i in selected_steps]
+                lines_to_keep = df_single_run[(df_single_run["key"] == metric) &
+                                              (df_single_run["step"].isin(steps_to_keep))]
+                result_df = pd.concat([result_df, lines_to_keep], ignore_index=True)
+        return result_df
+
+    def update_periodic_job(self, job_name, last_execution):
```

Comment:
Should be in the same session of sample_oldest_metrics. If one of the 2 failed, everything must be rollbacked